### PR TITLE
Publish website on push

### PR DESCRIPTION
**Pull request by satabin**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -55,6 +57,35 @@ ThisBuild / githubWorkflowPublish := Seq(
...
+
+ThisBuild / githubWorkflowAddedJobs += WorkflowJob(
+  id = "site",
+  name = "Deploy site",
+  needs = List("publish"),
+  javas = (ThisBuild / githubWorkflowJavaVersions).value.toList,
+  scalas = (ThisBuild / scalaVersion).value :: Nil,
+  cond = """
+  | always() &&
+  | needs.build.result == 'success' &&
```
Comment (ybasket) [inline build.sbt:70]: Couldn't you just remove these two lines? `always()` is to have this job run even when the dependencies failed, but you rule that out in the next line.

Comment (satabin) [reply build.sbt:70]: Copy paste from the fs2 repo 😅 I'll remove it, you're right.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -126,4 +126,45 @@ jobs:
...
           SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
           SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
-        run: sbt --client '++${{ matrix.scala }}; ci-release'
\ No newline at end of file
+        run: sbt --client '++${{ matrix.scala }}; ci-release'
+
+  site:
+    name: Deploy site
+    needs: [publish]
+    if: needs.build.result == 'success' && (needs.publish.result == 'success' && startsWith(github.ref, 'refs/tags/v'))
```
Comment (ybasket) [inline .github/workflows/ci.yml:134]: It doesn't even define a `build` need. And anyway, GH will only execute if the needs were successful, so:
```suggestion
    if: startsWith(github.ref, 'refs/tags/v')
```

File: build.sbt [hunk 3]
```diff
@@ -55,6 +57,34 @@ ThisBuild / githubWorkflowPublish := Seq(
...
+ThisBuild / githubWorkflowAddedJobs += WorkflowJob(
+  id = "site",
+  name = "Deploy site",
+  needs = List("publish"),
+  javas = (ThisBuild / githubWorkflowJavaVersions).value.toList,
+  scalas = (ThisBuild / scalaVersion).value :: Nil,
+  cond = """
+  | needs.build.result == 'success' &&
+  | (needs.publish.result == 'success' && startsWith(github.ref, 'refs/tags/v'))
+  """.stripMargin.trim.linesIterator.mkString.some,
```
Comment (ybasket) [inline build.sbt:71]: Ah, the suggestion from above should have been here:
```suggestion
 cons = "startsWith(github.ref, 'refs/tags/v')"
```

**Review by ybasket — APPROVED** [review]

**Pull request merged**